evm/gas-service.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const { getDefaultProvider, Contract } = ethers;
+const { Command, Option } = require('commander');
+const {
+    printInfo,
+    printWalletInfo,
+    printWarn,
+    mainProcessor,
+    prompt,
+    getContractJSON,
+    getGasOptions,
+    wasEventEmitted,
+    isValidAddress,
+    validateParameters,
+    httpPost,
+    loadConfig,
+} = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet } = require('./sign-utils');
+
+async function getGasUpdates(env, chain, destinationChains) {
+    const config = loadConfig(env);
+    const api = config.axelar.api;
+
+    return Promise.all(
+        destinationChains.map(async (destinationChain) => {
+            const destinationConfig = config.chains[destinationChain];
+
+            if (!destinationConfig) {
+                printError(`Error: chain ${destinationChain} not found in config.`);
+                printError(`Skipping ${destinationChain}.`);
+                return null;
+            }
+
+            const { gasEstimationType = 0, blobBaseFee = 0 } = destinationConfig;
+
+            const data = await httpPost(`${api}/gmp/getFees`, {
+                sourceChain: chain.axelarId,
+                destinationChain,
+                sourceTokenAddress: '0x0000000000000000000000000000000000000000',
+            });
+
+            const {
+                source_base_fee: sourceBaseFee,
+                source_token: {
+                    gas_price_in_units: { value: gasPrice },
+                    token_price: { usd: srcTokenPrice },
+                    decimals,
+                },
+                destination_native_token: {
+                    token_price: { usd: destinationTokenPrice },
+                },
+                execute_gas_multiplier: multiplier = 1.1,
+            } = data.result;
+
+            const axelarBaseFee = Math.ceil(parseFloat(sourceBaseFee) * Math.pow(10, decimals));
```

Comment:
Are you sure you don't need to use BigNumbers here? (18 decimals)

Reply:
I was able to implement it with FixedNumber instead, but they have lower precision only of 10 decimals.
I think it's actually better to use floats for the calculation. And rounded float is converted to BN automatically.